[AppServices] [OLD] Disallow creating runtime and scripted fields with * in the name (#110081)

shivindera opened the pull request:
## Summary

Disallow creating runtime and scripted fields with * in the name #110081.

Runtime field
![Screenshot_20211018_123027](https://user-images.githubusercontent.com/5311338/137714563-3047e128-4ccf-4cb8-a52f-8daf23802c0e.png)

Scripted field
![Screenshot_20211018_123130](https://user-images.githubusercontent.com/5311338/137714760-c8918ca2-d0e1-4a2e-aa36-bdfc2a227d52.png)


### Checklist

Delete any items that are not applicable to this PR.

- [x] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/README.md)
- [x] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios
- [x] This renders correctly on smaller devices using a responsive layout. (You can test this [in your browser](https://www.browserstack.com/guide/responsive-testing-on-local-server))


### For maintainers

- [x] This was checked for breaking API changes and was [labeled appropriately](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)

shivindera committed:
[AppServices] Disallow creating runtime and scripted fields with * in the name (#110081)

shivindera committed:
fixes for PR #115334

shivindera committed:
Merge remote-tracking branch 'upstream/master' into disallow-star-in-field

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/163225/)
* Commit: e390f9bf375bc7105465fc78f798d15c3ba1a03a
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Check i18n](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/163225/execution/node/430/log)

### History
* :broken_heart: [Build #161955](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/161955/) failed 245b453cc406a1cc93cc1e69bed6c630198d68a9

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/161955/","number":"161955","commit":"245b453cc406a1cc93cc1e69bed6c630198d68a9"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/163225/","number":"163225","commit":"e390f9bf375bc7105465fc78f798d15c3ba1a03a"}]}
PIPELINE-->

shivindera committed:
fix i18n namespace error in star_character_check file

mattkime reviewed (approved):
Congrats on your first PR! Works well and the code changes look good

loixlab reviewed (changes requested):
Code looks great except for the extra form validator. As mentioned in my comment there is no need to create a new one as it can easily be created out of the existing `containsChar` validator.

shivindera committed:
remove star_character_check to use contains_char for PR #115334

loixlab reviewed (approved):
Thanks for making the change, LGTM! 👍 
Left a comment for a leftover.

kibanamachine committed:
Merge branch 'master' into disallow-star-in-field

Commit:
[RAC][Observability] Use no data screen for the alerts and cases pages (#115178)

Co-authored-by: Chris Cowan <chris@chriscowan.us>
Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

machadoum committed:
Standardize page header layouts in Security (#115393)

* Standardize page header layouts in Security

* Update Observability create case layout

mistic committed:
skip flaky suite (#115529)

mistic committed:
skip flaky suite (#115883)

pheyos committed:
[ML] Functional tests - re-activate a11y suite (#115900)

This PR re-activates the ML accessibility test suite.

mistic committed:
skip flaky suite (#115320)

loixlab committed:
[UA] Align server config on master

mshustov committed:
remove ngreact. no angular in the codebase (#115895)

VladLasitsa committed:
[TSVB] Chart with override index pattern and last_value are migrated with entire_time_range from 7.13+ (#115041)

* Move logic about default time range mode inside index pattern component

* Fix lint

* Fix types

* Cover case with override_index_pattern

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

jloleysens committed:
[Reporting] Baseline capture tests (#113910)

* added page to reporting example app that contains the capture tests

* first version of PNG capture for test A

* added types file to common

* added data-shared-item attr to image, also added capture menu items

* fix image CSS by providing a fixed width and height

* explicitly add layout for print, does not seem to do anything though?

* added magic numbers of image sizes

* added reporting examples test folder

* first version of capture test for generating and comparing PNGs

* added PNG service and PNG baseline fixture

* added pdf-to-img dev dependency

* refactor compare_pngs to accept a buffer

* added comment to interface

* png service -> compare images service

* export image compare service

* added test for pdf export

* clean up log

* minor fixes and added pdf print optimized test

* added pdf and pdf print fixtures

* refactor lib function name

* Update difference thresholds

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

stratoula committed:
[Pie] Fixes the deprecation message on the legacy library switch (#115915)

shahzad31 committed:
[Uptime]Synthetics step fix browser console indexing error (#115897)

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

nickofthyme committed:
update elastic-charts to 38.0.1 (#115623)

awahab07 committed:
[uptime] Adjust the theme colors for uptime monitors browser screenshot labels. (#115543)

* Add useBreakpoints reactive hook for uptime. Adjusted color/contrast and alignment for timestamp screenshot popup footer.

#96122

MadameSheema committed:
updates Error legend default message to align with tooltip message and the other legend texts (#115896)

mikecote committed:
Initial commit (#115846)

dokmic committed:
[Reporting] Use updated saved object resolver (#113641)

* Update deprecated PDF job factory to use updated saved object resolver
* Reuse data plugin service to gather index pattern attributes in the deprecated CSV export type

jloleysens committed:
[Reporting] Add guidance when copying URLs that may have issues (#112979)

* --wip-- v1 of the UX [skip-ci]

* copy update [skip-ci]

* also respect the max URL limit

* use "isUnsaved"

* move files around and introduce callout for warnings

* remove proceed anyway

* cleaned up imports

* fix i18n and types

* added jest tests and update jest snapshot

* rewrite max length for reader clarity.

* refactor unsaved "warning" to unsaved "error" and do not show the copy URL button

* update screenshot capture panel test and snapshots

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

TinaHeiligers committed:
Fix flaky functional test in saved objects managements -> delete SO (#115291)

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

weltenwort committed:
[Logs UI] [Metrics UI] Register deprecations for the logAlias and metricAlias config settings (#115845)

pheyos committed:
[ML] Functional tests - stabilize categorization examples endpoint test (#115910)

This PR stabilizes the categorization example endpoint tests by using a different esArchive.

yuliacech committed:
[Index Management] Forward port of the fix for indices table test in 7.16 (#115933)

alexwizp committed:
[CodeEditor] add support of triple quotes (#112656)

* [CodeEditor] add support of triple quotes

* add tests for grammar

* an escaped quote can be appended to the end of triple quotes'

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

FrankHassanabad committed:
Fixes backwards compatibility with rule exports (#115888)

## Summary

We at least want to do our best to retain n-1 version compatible with exports and then best effort for n-2 or more. During testing I saw that we changed our export contract and I was getting these errors importing from 7.15 into 7.16:

<img width="814" alt="Screen Shot 2021-10-20 at 8 25 07 PM" src="https://user-images.githubusercontent.com/1151048/138202992-c6e016e6-388c-4597-b302-b13d7126e705.png">

The reason is because it thinks the final "summary line" which has `exported_count` instead of `exported_rules_count` will think that is a rule based ndjson line and not exclude it.

This was introduced in this PR: https://github.com/elastic/kibana/pull/115144

I was under the impression that was adding more information and not subtracting information when I did the PR review. I almost think that we should keep the `exported_count` in addition to the new lines that are added so the user has a total count of everything exported. This PR doesn't do that, it just fixes the backwards compatibility to not have errors. If we ship like this the other rules from 7.15 and previous will still import but users will see error messages like that one above which they should never see.

However, really, we should be careful about changes that are not just additive with formats. 

### Checklist

No this PR currently does not have tests for backwards compatibility but I think we should add it either here or a follow up.

- [ ] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios

FrankHassanabad committed:
Removes logger errors that are not needed (#115866)

## Summary

When moving the saved object references I added logger errors and utilities to detect when the original data structure id changed from the saved object reference id's. However, when things move to mulit-space or if users upload different id's within these structures from something such as the saved object management system then users will see these errors in their logs when they shouldn't. I  removed them all as this is expected behavior that the references array can and will contain different id's.

Before you would see these messages on an upgrade to 8.0.0 or if you uploaded or changed a rule id through the SOM:

```
[2021-10-18T17:36:03.550-06:00][ERROR][plugins.securitySolution] The id of the "saved object reference id": ec94193b-4788-5ceb-b8fb-5e270836beec is not the same as the "saved object id": 6caeff30-3069-11ec-83d2-0376356fe525. Preferring and using the "saved object reference id" instead of the "saved object id"
```

Now you will not.


### Checklist
- [x] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios

paul-tavares committed:
Fix label for fields in the Advanced settings section of policy details (#115869)

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

dej611 committed:
[Lens] Do not show legend actions popup when in non-interactive mode (#115804)

* :bug: Fix non-interactive legend issue

* :white_check_mark: Add tests

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

dej611 committed:
[Lens] Relax break down group validation for percentage charts (#114803)

* :bug: Improved percentage checks for breakdown group

* :white_check_mark: Add tests

* :ok_hand: Integrate feedback

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

lcawl committed:
Fixes python URL in documentation link service (#115886)

hop-dev committed:
[Fleet] Link to the installed version of an integration from global search (#115736)

* remove version from integrations global search link

* fix tests

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

bhavyarm committed:
Unskip dashboard and dashboard panel a11y tests (#115102)

lukasolson committed:
Use extract/inject in saved query service (#111213)

* Use extract/inject in saved query service

* Add route handler context

* Allow query objects & update tests

* Add migrations to saved query type

* Review feedback

* Review feedback

* Separate create/update methods

* Fix copy/paste bug

* Update jest snapshot

hop-dev committed:
remove hash router related code (#115733)

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

kevinlog committed:
[Security Solution] Fix metadata tests with updated timestamps (#115591)

bhavyarm committed:
unskip kibana overview test from xpack and add the missing painless labs test to config.ts (#115240)

byronhulcher committed:
[App Search] Improve loading experience on the Curation and Curation Detail pages (#115840)

byronhulcher committed:
[App Search] Restrict log stream on Automated Curation Detail page to only automated events (#115841)

qn895 committed:
[ML] Add Index data visualizer grid embeddable as extra view within Discover (#107184)

* [ML] Initial embed

* [ML] Initial embed props

* [ML] Add top nav link to data viz

* Add visible fields

* Add add data service to register links

* Renames, refactor, use constants

* Renames, refactor, use constants

* Update tests and mocks

* Embeddable

* Update hook to update upon time udpate

* Add filter support to query

* Refactor filter utilities

* Add filter support for embeddable

* Fix saved search data undefined

* Prototype aggregated view/document view switcher

* Prototype flyout

* Prototype save document view option in storage

* Fix filter and query conflict with saved search

* Minor styling edits

* [ML] Initial embed

* [ML] Initial embed props

* Add embeddable 1

* Add visible fields

* Embeddable 2

* Add filter support to query

* Refactor filter utilities

* Add filter support for embeddable

* Fix saved search data undefined

* Prototype aggregated view/document view switcher

* Prototype flyout

* Prototype save document view option in storage

* Fix filter and query conflict with saved search

* Minor styling edits

* Fix missing code after conflicts

* Remove dv locator and flyout

* Make types happy

* Fix types

* Rename toggle option

* Resolve conflicts

* [ML] Reduce size of chart

* [ML] Unbold name, switch icons of show distributions

* [ML] Make size consistent

* [ML] Make page size 25

* [ML] Switch to arrow right and down

* [ML] Make legend font smaller

* [ML] Add user setting

* [ML] Add show preview by default setting

* [ML] Match icon

* Add panels around the subcontent

* Add preference for aggregated vs doc

* Fix types

* Fix types, add constants for adv settings

* Change to data view type

* Temp fix for Kibana/EUI table overflow issue

* Modify line height so text is not cut off, modify widths for varying screen sizes

* Different width padders for different screens

* Fix CI

* Merge latest, move button to the right

* Fix width for bar charts previews

* Fix toggle buttons, fix maps

* Delete unused file

* Fix boolean styling

* Change to enum, discover mode

* Hide field stats

* Hide field stats

* Persist show mini preview/distribution settings

* Remove window size, use size observer instead

* Default to document view

* Remove bold, switch icon

* Set fixed width for top values, reduce font size in table

* Fix custom url tests

* Update width styling for panels

* Fix missing flag for Discover sidebar, jest tests

* Fix max width

* Workaround for sorting

* Fix import

* Fix styling

* Make height uniform, center alignment, fix map and keyword map not same size

Move styling

* Revert "Make height uniform, center alignment, fix map and keyword map not same size"

This reverts commit 8fc42e2f

* Revert "Make height uniform, center alignment, fix map and keyword map not same size"

This reverts commit 8fc42e2f

* Uniform height, left aligned, flex grid

* Switch top values to have labels

* Center content

* Replace fixed widths with percentage

* Fix table missing field types

* Add dashboard embeddable and filter support

* Fix file data viz styling and tests, lean up imports, remove hard coded pixels

* Add search panel/kql filter bar

* Temporarily fix scrolling

* New kql filters for data visualizer

* Set map height so it will fit the sampler shard size text

* Use eui progress labels

* Fix spacer

* Add beta badge

* Temporarily fix scrolling

* Fix grow for Top Values for

* [ML] Update functional tests to reflect new arrow icons

* [ML] Add filter buttons and KQL bars

* [ML] Update filter bar onChange behavior

* [ML] Update top values filter onChange behavior

* [ML] Update search filters when opening saved search

* [ML] Clean up

* [ML] Remove fit content for height

* [ML] Fix boolean legend

* [ML] Fix header section when browser width is small to large and when index pattern title is too large

* [ML] Hide expander icon when dimension is xs or s & css fixes

* [ML] Delete embeddables because they are not use

* [ML] Rename view mode, refactor to separate hook, add error prompt if can't show, rename wrapper, clean up & fix tests

* [ML] Make doc count 0 for empty fields, update t/f test

* [ML] Add unit testing for search utils

* Fix missing unsubscribe for embeddable output

* Remove redundant onAddFilter for this PR, fix width

* Rename Field Stats to Field stats to match convention

* [ML] Fix expand all/collapse all behavior to override individual setting

* [ML] Fix functional tests should be 0/0%

* [ML] Fix docs content spacing, rename classnames, add filters to Discover, lens, and maps

* [ML] Fix doc count for fields that exists but have no stats

* [ML] Fix icon styling to match Discover but have text/keyword/histogram

* [ML] Fix doc count for fields that exists but have no stats

* [ML] Rename classnames to BEM style

* Resolve latest changes

* [ML] Add tests for data viz in Discover

* Update tests & dashboard behavior to reflect new advanced settings

* Update telemetry

* Remove workaround after eui bump fix

* Fix missing bool clause

* Add login

* Fix saved search attributes broken with latest changes

* Update tests

* Fix import

* Match the no results found

* Fix query util to return search source's results right away. Fix texts.

* Rename old test suits to farequoteDataViewTestData

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

mistic committed:
fix(NA): creation of multiple processes on production by splitting no_transpilation when setting up node env (#115246)

* fix(NA): adds no_transpilation_dist to avoid preserve_symlinks on dist

* chore(NA): setup node env correctly on functional tests

* chore(NA): try to fix tests

* chore(NA): correctly separate split

* chore(NA): check ensure preserve symlinks need

* chore(NA): investigate path resolve result

* chore(NA): investigate path resolve result #2

* chore(NA): comment out preserve symlinks

* chore(NA): apply fs.realpathSync into the calculated REPO_ROOT paths on babel_register_for_test_plugins

* chore(NA): removes debug code

* chore(NA): move array definition

* chore(NA): correctly import fs

* chore(NA): add debug code

* chore(NA): some more debug statements

* chore(NA): remove ensure symlinks

* chore(NA): trying to solve double symlinking

* chore(NA): test mappings

* chore(NA): process path

* chore(NA): test a second map

* chore(NA): using a different mappings

* chore(NA): more debug cases

* chore(NA): more debug logic

* chore(NA): more debug cases

* chore(NA): more debug cases

* chore(NA): more debug cases

* chore(NA): try to add realpathSync into require

* chore(NA): try to add realpathSync into require

* fix(NA): jenkins and buildkite run

* chore(NA): add debug logs

* chore(NA): correct path

* chore(NA): correct path

* chore(NA): add more test maps

* chore(NA): add more test maps

* chore(NA): add some more test maps experiments

* chore(NA): try to remove another test map dep

* chore(NA): try to remove another test map dep

* chore(NA): try to remove another test map dep

* chore(NA): try to remove another test map dep

* chore(NA): try to remove another test map dep

* chore(NA): try to remove another test map dep

* chore(NA): try to remove another test map dep

* chore(NA): try to remove another test map dep

* chore(NA): include all correct transpilations for each jenkins path

* chore(NA): include all correct transpilations for each used asset path

* chore(NA): include all correct transpilations for each used asset path

* chore(NA): remove jenkins support

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

mbondyra committed:
[Lens] Fix editor blowing up when working on non-exisiting data view (#114816)

* correct styles for config panel (if data view is unavailable, the margins are still ok)

* temp

* fix data views bugs

* add test

* integrate feedback

* Update datapanel.tsx

* Update x-pack/plugins/lens/public/indexpattern_datasource/loader.ts

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

juliaElastic committed:
[Fleet] remove old assets for manual upgrade too (#115926)

* remove old assets for manual upgarde too

* fixed tests

* fixed tests

nreese committed:
[Maps] fix locked tooltip issues (#115583)

* [Maps] fix locked tooltip issues

* rename cleanTooltipStateForLayer to updateTooltipStateForLayer

* remove duplicated return

* remove commented out line

* review feedback

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

joshdover committed:
[Fleet] Update beats tutorial descriptions for unified integrations (#115829)

Co-authored-by: gchaps <33642766+gchaps@users.noreply.github.com>
Co-authored-by: DeDe Morton <dede.morton@elastic.co>

paul-tavares committed:
[Security Solution][Endpoint] un-skip permissions FTR tests for endpoint (#115962)

* unskip test suite and skip the Test for Host details - found bug

YulNaumenko committed:
[Alerting] Changed alerting telemetry throttle and schedule time metrics to be numbers instead of strings (#115605)

* [Alerting] Changed alerting telemetry throttle and interval metrics to be numbers instead of strings

* fixed test

* fixed typecheck

* set size to 0

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

cauemarcondes committed:
[APM] Dark metadata icons on service overview page (#115964)

chrisronline committed:
Add deprecation levels (#115832)

alvarezmelissa87 committed:
[FLEET] Increase asset size limit for installing ML model packages (#115890)

* update size limit for assets to 50mb

* use different size limit for ml model

* add byte constant for clarity

phillipb committed:
[Stack Monitoring] Fix missing breadcrumbs to logstash (#115780)

kevinlog committed:
[Security Solution] Adds missing migration script execution for Endpoint policy (#115969)

mbondyra committed:
[Lens] cleanup divide mock file to smaller pieces (#115925)

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

snide committed:
Copy changes for sample data header (#115963)

poffdeluxe committed:
[Controls] Data view and field pickers (#116018)

Added file picker and field picker to presentationUtil

mistic committed:
chore(NA): upgrades lmdb-store to v1.6.11 (#115971)

* chore(NA): upgrades lmdb-store to v1.6.10

* chore(NA): upgrade into v1.6.11

YulNaumenko committed:
[Alerting] Telemetry fix for min/max number of actions a rule has associated with (#115496)

* [Alerting] Telemetry for max number of actions a rule has and max alerts created on execution

* -

* Update rules_client.ts

* fixed task data

* added unit test

* fixed by adding runtime field

* fixed task data

* fixed test

* fixed telemetry for throttle and interval

* fixed task data

* fixed task data

* fixed test

* fixed due to comments

* fixed typecheck

* fixed test

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

rashmivkulkarni committed:
Removing tests that support legacy exports (#116011)

* deleted 2 files

* removing the references of the deleted file

lukeelmers committed:
Remove migrations.enableV2 in 8.0 (#116023)

neptunian committed:
don't request when request is pending (#115999)

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

dgieselaar committed:
[APM] Add live mode to synthtrace (#115988)

cnasikas committed:
[Connectors] Check connector's responses (#115797)

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

VladLasitsa committed:
[TSVB] Error when selecting * for override index_pattern field on the string mode (#114450)

* Add condition so that preselect first timefield from the list if we use * as index pattern

* Add default timefield for string index pattern mode

* Add comment for condition

* Fix lint

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

gbamparop committed:
Update APM linting dev doc with instructions about how to install the pre-commit hook and a link to the Kibana guide (#115924)

phillipb committed:
Fix bug with popup not closing (#115954)

walterra committed:
[ML] Fix legend text colors for Vega based charts in dark mode. (#115911)

Fixes the legend text colors for Vega based charts in dark mode.

angorayc committed:
[Security Solution] Fix tooltip for timeline sourcerer (#115950)

* fix tooltip for event picker

* remove unused mock

* remove unused dependency

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

dimaanj committed:
[Discover] Add clock icon to time field column (#114996)

* [Discover] Add clock icon to time field column

* [Discover] fix linting

* [Discover] fix functional tests

* [Discover] fix functional test

* [Discover] apply suggestion

* [Discover] apply wording suggestion

* [Discover] fix i18n

* [Discover] update unit tests

* [Discover] add custom label for classic table

* [Discover] color icon black

* [Discover] fix unit tests

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

shivindera committed:
add fix for a failing lens test & remove unused type
